pyriemann/classification.py

```diff
@@ -863,36 +887,120 @@ class MeanField(SpdClassifMixin, TransformerMixin, BaseEstimator):
+        power_mean_maxiter=150,
+        distance_squared=True,
+        reuse_previous_mean=False,
         n_jobs=1,
     ):
         """Init."""
         self.power_list = power_list
         self.method_label = method_label
         self.metric = metric
+        self.power_mean_zeta = power_mean_zeta
+        self.power_mean_maxiter = power_mean_maxiter
+        self.distance_squared = distance_squared
+        self.reuse_previous_mean = reuse_previous_mean
         self.n_jobs = n_jobs
 
+        if self.reuse_previous_mean and self.n_jobs != 1:
+            raise Exception(("Currently reuse_previous_mean is not supported "
+                            "when combined with parallel mean calculation."))
+
+        if method_label not in {'lda', 'sum_means', 'inf_means'}:
+            raise TypeError("Parameter method_label is incorrect.")
+
+        if self.method_label == "lda":
+            self.lda = LDA()
+
+    def _calculate_mean(self, X, y, p, sample_weight):
+        '''
+        Calculates power means for all classes for specific p.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_matrices, n_channels, n_channels)
+            Set of SPD matrices.
+        y : ndarray, shape (n_matrices,)
+            Labels for each matrix.
+        p : float
+            Exponent of a power mean.
+        sample_weight : None | ndarray shape (n_matrices,), default=None
+            Weights for each matrix. If None, it uses equal weights.
+
+        Returns
+        -------
+        means_p : dict
+            Contains the power means for all classes for this p.
+
+        '''
+        means_p = {}
+
+        for ll in self.classes_:
+
+            init = None
+
+            if self.reuse_previous_mean and self.n_jobs == 1:
+                pos = self.power_list.index(p)
+                if pos > 0:
+                    prev_p = self.power_list[pos-1]
+                    if prev_p in self.covmeans_:
+                        init = self.covmeans_[prev_p][ll]
+                    else:
+                        raise Exception("No previous mean.")
+
+            means_p[ll] = mean_power(
+                X[y == ll],
+                p,
+                sample_weight=sample_weight[y == ll],
+                zeta=self.power_mean_zeta,
+                init=init,
+                maxiter=self.power_mean_maxiter
+            )
+
+        return means_p
+
+    def _calculate_all_means(self, X, y, sample_weight):
+
+        if self.n_jobs == -1 or self.n_jobs > 1:
+            results = Parallel(n_jobs=-1)(
+                delayed(self._calculate_mean)(X, y, p, sample_weight)
+                for p in self.power_list
+            )
+
+            for i, p in enumerate(self.power_list):
+                self.covmeans_[p] = results[i]
+        else:
+            for p in self.power_list:
+                result_per_p = self._calculate_mean(X, y, p, sample_weight)
+                self.covmeans_[p] = result_per_p
```

Comment:
Joblib is already configured to work with 1 job. See:

![image](https://github.com/user-attachments/assets/e09a5dfe-eef8-4ce3-9afd-96d1b68a4075)

https://joblib.readthedocs.io/en/latest/generated/joblib.Parallel.html